crates/hyperqueue/src/transfer/messages.rs

```diff
@@ -92,12 +92,19 @@ pub struct SubmitRequest {
 }
 
 #[derive(Serialize, Deserialize, Debug, Clone)]
-pub enum Selector {
+pub enum IdSelector {
     All,
     LastN(u32),
     Specific(IntArray),
 }
 
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub enum StatusSelector {
```

Comment:
Now that I look at it, `None` is probably not a good choice here. It would allow you to send e.g. `IdSelector::All` and `StatusSelector::None`, which doesn't really make much sense. We could optimize this situation in `make_job_detail`, but it shouldn't be possible to ask for this in the first place. And it wouldn't be clear if the client wants to warn about missing IDs in such case or not.

Let's introduce something like `TaskSelector`, which will contain both `IdSelector` and `StatusSelector` fields. Then remove the `None` variant from `StatusSelector` and add `Option<TaskSelector>` to `JobDetailRequest`. Not requiring tasks would then be modeled as passing `None` to it.